polynote-kernel/src/main/scala/polynote/kernel/environment/environment.scala

```diff
@@ -237,6 +237,36 @@ object InterpreterEnvironment {
   */
 object Env {
 
+
+  def addM[RO]: AddMPartial[RO] = addMPartialInst.asInstanceOf[AddMPartial[RO]]
```

Comment:
Might be good to have some comments for this section here. Have to admit I'm not too sure I follow what's going on. 

What's the benefit of using `addM` over `enrichWith`? 

I see that you replaced
```
Env.enrichWith[BaseEnv, GlobalEnv](Environment, GlobalEnv(config, interps, kernelFactory))
```
with 
```
Env.addM[BaseEnv](ZIO.succeed(GlobalEnv(config, interps, kernelFactory)))
```
but I'm not too clear on the difference between the two. I guess `addM` is a bit cleaner because we don't have to restate `GlobalEnv` as a type and we don't need `Environment`. Is that essentially the idea or is there more to it that I'm missing?

Reply:
Also, in the above case it looks like it could be further simplified into `Env.add[BaseEnv](GlobalEnv(config, interps, kernelFactory))`?